### DISCUSSION TITLE
Want to transfer a value of a smart meter to a var using it later within the script

### DISCUSSION THREAD

**Discussion by Sfre12** [arendst/Tasmota#16224] [Q&A] [answered]
Hello all.
I am working with Tasmota V12.0.4 and SML support to read my smartmeter
Right now I am working on a device which is connected to my smart meter and should control some relay`s.
The code from the example here:https://tasmota.github.io/docs/Smart-Meter-Interface/#hager-ehz363-sml-with-daily-values is working well without changing anything instead of adding two “print” commands.
Now I want to transfer the content of the var “Total_in” to use by script and tried to transfer it with command 
pin=SML# Total_in
as it is in the example
However, I have an output in WebUI, but the var is still Zero
How is it possible to transfer the content of the smart meter to a var I can use with the script?
![Unbenannt-1](https://user-images.githubusercontent.com/79339733/184553001-415a4866-a089-467a-b421-f42f783b4d46.jpg)

**Comment by hagenbuch** [comment]
I'm having the same problem (Tasmota 11.1.0, bought a preflashed device)

Could you maybe post your >M meter section so we see if "SML" is really your counter identification and "Total_in" are defined there as well?

Andreas

**Reply by barbudor** [reply]
May be the key is one of:
```
sml[n] = get value of SML energy register n (if defined USE_SML_SCRIPT_CMD)
enrg[n] = get value of energy register n 0=total, 1..3 voltage of phase 1..3, 4..6 current of phase 1..3, 7..9 power of phase 1..3 (if defined USE_ENERGY_SENSOR)
```

**Comment by Zumili** [comment, marked as answer]
Like @barbudor already said, you need to define `USE_SML_SCRIPT_CMD` in **my_user_config.h** or **user_config_override.h**
`#define USE_SML_SCRIPT_CMD`
In newer version of Tasmota, this is now defined by default.
Then you can do something like this:
```
>D
p:total_in=0
p:total_out=0
current_consumption=0
>B
->sensor53 r
>M 1
+1,3,s,0,9600,SML
1,77070100010800ff@1000,Total consumption,KWh,Total_in,4
1,77070100020800ff@1000,Total Feed,KWh,Total_out,4
1,77070100100700ff@1,Current consumption,W,Power_curr,0
;... more lines ...
>S
total_in=sml[1]
total_out=sml[2]
current_consumption=sml[3]
print Total in: %total_in%
print Total out: %total_out%
print Current consumption: %current_consumption%
#
```
Keep in mind, that the order of the lines in the **>M** section define the indices of the **sml-array**.

**Reply by Sfre12** [reply]
Thank you all!!
This is working
